server/src/main/java/org/candlepin/model/dto/ProductContentData.java

```diff
@@ -0,0 +1,238 @@
+/**
+ * Copyright (c) 2009 - 2016 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.model.dto;
+
+import org.candlepin.model.Content;
+import org.candlepin.model.ProductContent;
+
+import io.swagger.annotations.ApiModel;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+import javax.xml.bind.annotation.XmlRootElement;
+
+
+
+/**
+ * DTO representing a product content mapping exposed to the API and adapter layers. Unlike the
+ * ProductContent entity, this DTO does not contain a reference to the product. Instead, it is
+ * implied by attaching it to a ProductData instance.
+ *
+ * <pre>
+ * {
+ *   "content" : {
+ *     ...
+ *   },
+ *   "enabled" : false
+ * }
+ * </pre>
+ */
+@ApiModel(parent = CandlepinDTO.class)
+@XmlRootElement
+public class ProductContentData implements Cloneable {
+    private ContentData content;
+    private Boolean enabled;
+
+    /**
+     * Initializes a new ProductContentData instance with null values.
+     */
+    public ProductContentData() {
+        // Intentionally left empty
+    }
+
+    public ProductContentData(ContentData content, Boolean enabled) {
+        this.setContent(content);
+        this.setEnabled(enabled);
+    }
+
+    /**
+     * Initializes a new ProductContentData instance using the data contained by the given DTO.
+     *
+     * @param source
+     *  The source DTO from which to copy data
+     */
+    public ProductContentData(ProductContentData source) {
+        if (source != null) {
+            this.populate(source);
+        }
+    }
+
+    /**
+     * Initializes a new ProductContentData instance using the data contained by the given entity.
+     *
+     * @param entity
+     *  The source entity from which to copy data
+     */
+    public ProductContentData(ProductContent entity) {
+        if (entity != null) {
+            this.populate(entity);
+        }
+    }
```

Comment:
Not sure what you mean by "old with the null object." Is that supposed to be "okay with?" I'm going to answer assuming that's the case

Initially they did throw an exception much like the populate method itself. At some point I decided that if you intended to create a new DTO, you were probably intending to create it with whatever data you happened to have available. In the event of a null, that would be no data.

I suppose, writing it out, it doesn't make much sense. I can revert that change so the DTO constructors also throw an illegal argument exception in the event of a null entity or source DTO.